From 9fb038ca64ac33ea4135e0e810e391d93d5a71c5 Mon Sep 17 00:00:00 2001
From: Hannu Nyman <hannu.nyman@iki.fi>
Date: Sun, 3 Oct 2021 21:23:04 +0300
Subject: [PATCH 56/61] busybox: update to 1.34.1

Update busybox to version 1.34.1, which is a minor
maintenance release. It contains just the two post-1.34.0
upstream patches that we earlier backported plus a few fixes
to awk.

* Remove the two backported upstream patches that are
  now unnecessary.

Signed-off-by: Hannu Nyman <hannu.nyman@iki.fi>
---
 package/utils/busybox/Makefile                |  4 +-
 ...-backport-fix-build-breakage-on-mips.patch | 37 ---------
 ...02-backport-ash-process-substitution.patch | 77 -------------------
 3 files changed, 2 insertions(+), 116 deletions(-)
 delete mode 100644 package/utils/busybox/patches/001-backport-fix-build-breakage-on-mips.patch
 delete mode 100644 package/utils/busybox/patches/002-backport-ash-process-substitution.patch

diff --git a/package/utils/busybox/Makefile b/package/utils/busybox/Makefile
index 306da6c36c..90d323b92b 100644
--- a/package/utils/busybox/Makefile
+++ b/package/utils/busybox/Makefile
@@ -5,14 +5,14 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=busybox
-PKG_VERSION:=1.34.0
+PKG_VERSION:=1.34.1
 PKG_RELEASE:=$(AUTORELEASE)
 PKG_FLAGS:=essential
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=https://www.busybox.net/downloads \
 		http://sources.buildroot.net
-PKG_HASH:=ec8d1615edb045b83b81966604759c4d4ac921434ab4011da604f629c06074ce
+PKG_HASH:=415fbd89e5344c96acf449d94a6f956dbed62e18e835fc83e064db33a34bd549
 
 PKG_BUILD_DEPENDS:=BUSYBOX_CONFIG_PAM:libpam
 PKG_BUILD_PARALLEL:=1
diff --git a/package/utils/busybox/patches/001-backport-fix-build-breakage-on-mips.patch b/package/utils/busybox/patches/001-backport-fix-build-breakage-on-mips.patch
deleted file mode 100644
index 354e3e511a..0000000000
--- a/package/utils/busybox/patches/001-backport-fix-build-breakage-on-mips.patch
+++ /dev/null
@@ -1,37 +0,0 @@
-From 00adcdb64a5487f0ea6b400d912dcf7c58637696 Mon Sep 17 00:00:00 2001
-From: Denys Vlasenko <vda.linux@googlemail.com>
-Date: Fri, 20 Aug 2021 13:33:50 +0200
-Subject: udhcp: fix build breakage on MIPS
-
-Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
----
- networking/udhcp/common.h | 12 ------------
- 1 file changed, 12 deletions(-)
-
-diff --git a/networking/udhcp/common.h b/networking/udhcp/common.h
-index 8c678dd32..ca778dab8 100644
---- a/networking/udhcp/common.h
-+++ b/networking/udhcp/common.h
-@@ -304,18 +304,6 @@ void udhcp_dump_packet(struct dhcp_packet *packet) FAST_FUNC;
- # define log3s(msg) ((void)0)
- #endif
- 
--#if defined(__mips__)
--/*
-- * The 'simple' message functions have a negative impact on the size of the
-- * DHCP code when compiled for MIPS, so don't use them in this case.
-- */
--#define bb_simple_info_msg bb_info_msg
--#define bb_simple_error_msg bb_error_msg
--#define bb_simple_perror_msg_and_die bb_perror_msg_and_die
--#undef log1s
--#define log1s log1
--#endif
--
- /*** Other shared functions ***/
- 
- /* 2nd param is "uint32_t*" */
--- 
-cgit v1.2.3
-
-
diff --git a/package/utils/busybox/patches/002-backport-ash-process-substitution.patch b/package/utils/busybox/patches/002-backport-ash-process-substitution.patch
deleted file mode 100644
index 5b348ea26b..0000000000
--- a/package/utils/busybox/patches/002-backport-ash-process-substitution.patch
+++ /dev/null
@@ -1,77 +0,0 @@
-Stacy Harper reports that this script:
-
-   test() { . /tmp/bb_test; }
-   echo "export TEST=foo" >/tmp/bb_test
-   test 2>/dev/null
-   echo "$TEST"
-
-correctly prints 'foo' in BusyBox 1.33 but hangs in 1.34.
-
-Bisection suggested the problem was caused by commit a1b0d3856 (ash: add
-process substitution in bash-compatibility mode).  Removing the call to
-unwindredir() in cmdloop() introduced in that commit makes the script
-work again.
-
-Additionally, these examples of process substitution:
-
-   while true; do cat <(echo hi); done
-   f() { while true; do cat <(echo hi); done }
-   f
-
-result in running out of file descriptors.  This is a regression from
-v5 of the process substitution patch caused by changes to evalcommand()
-not being transferred to v6.
-
-function                                             old     new   delta
-static.pushredir                                       -      99     +99
-evalcommand                                         1729    1750     +21
-exitreset                                             69      86     +17
-cmdloop                                              372     365      -7
-unwindredir                                           28       -     -28
-pushredir                                            112       -    -112
-------------------------------------------------------------------------------
-(add/remove: 1/2 grow/shrink: 2/1 up/down: 137/-147)          Total: -10 bytes
-
-Signed-off-by: Ron Yorston <rmy at pobox.com>
----
- shell/ash.c | 10 +++++++---
- 1 file changed, 7 insertions(+), 3 deletions(-)
-
-diff --git a/shell/ash.c b/shell/ash.c
-index b5947147a..53c140930 100644
---- a/shell/ash.c
-+++ b/shell/ash.c
-@@ -10278,6 +10278,9 @@ evalcommand(union node *cmd, int flags)
- 
- 	/* First expand the arguments. */
- 	TRACE(("evalcommand(0x%lx, %d) called\n", (long)cmd, flags));
-+#if BASH_PROCESS_SUBST
-+	redir_stop = redirlist;
-+#endif
- 	file_stop = g_parsefile;
- 	back_exitstatus = 0;
- 
-@@ -10356,7 +10359,11 @@ evalcommand(union node *cmd, int flags)
- 		lastarg = nargv[-1];
- 
- 	expredir(cmd->ncmd.redirect);
-+#if !BASH_PROCESS_SUBST
- 	redir_stop = pushredir(cmd->ncmd.redirect);
-+#else
-+	pushredir(cmd->ncmd.redirect);
-+#endif
- 	preverrout_fd = 2;
- 	if (BASH_XTRACEFD && xflag) {
- 		/* NB: bash closes fd == $BASH_XTRACEFD when it is changed.
-@@ -13476,9 +13483,6 @@ cmdloop(int top)
- #if JOBS
- 		if (doing_jobctl)
- 			showjobs(SHOW_CHANGED|SHOW_STDERR);
--#endif
--#if BASH_PROCESS_SUBST
--		unwindredir(NULL);
- #endif
- 		inter = 0;
- 		if (iflag && top) {
--- 
-2.31.1
-- 
2.30.2

